code/modules/atmospherics/machinery/portable/canister.dm

```diff
@@ -99,6 +103,8 @@ GLOBAL_LIST_INIT(gas_id_to_canister, init_gas_id_to_canister())
 	else
 		create_gas()
 
+	update_window()
```

Comment:
Don't you already call update_appearance here? why do this too?

Reply:
Used to be update_window wasn't being called by update_appearance, but yeah this should be removed now

Reply:
Scratch that I kept this and handle the overlay manually in the window update. Better than running the whole update_appearance again.